docs/workforce-management-framework/topics/pro-keycloak-enablement.adoc

```diff
@@ -24,29 +28,117 @@ docker run -p 8080:8080 feedhenry/raincatcher-keycloak
+[discrete]
+== Importing the {Keycloak} Realm Settings
+
+NOTE: This step should only be done if you choose to run your own {Keycloak} instance. The {Keycloak} Docker image
+comes with a demo RainCatcher realm setup therefore it does not need to be manually imported.
+
+. Download the link:{WFM-RC-KeycloakDockerURL}{WFM-RC-Branch}/data_files/raincatcher-realm.json[{PROJECT_NAME} realm settings]
+. Login to the {Keycloak} admin console. See xref:{context}-accessing-the-keycloak-admin-console[Accessing the {Keycloak} Admin Console] for more information.
+. Click 'Add Realm'
+. Import the link:{WFM-RC-KeycloakDockerURL}{WFM-RC-Branch}/data_files/raincatcher-realm.json[{PROJECT_NAME} realm settings] you downloaded
+. Click 'Create'
+
+This will create the demo RainCatcher realm which includes the setup for three different clients: raincatcher-portal, raincatcher-mobile and raincatcher-cloud.
+
+IMPORTANT: The provided link:{WFM-RC-KeycloakDockerURL}{WFM-RC-Branch}/data_files/raincatcher-realm.json[{PROJECT_NAME} realm settings] is only used for
+demo purposes. Ensure that you provide proper client configurations for production environments. For more information 
+about client configurations, see the {Keycloak} link:{WFM-RC-KeycloakURL}server_admin/topics/clients.html[clients documentation]
+
+[id='{context}-create-own-keycloak-realm']
+[discrete]
+== Creating your own {Keycloak} Realm
+IMPORTANT: Ensure that you have a deep understanding of {Keycloak}, Node.js and Javascript Adapters before attempting to change any of the configurations below.
+
+NOTE: This step should only be done if you choose to create your own {Keycloak} realm instead of using the existing demo
+link:{WFM-RC-KeycloakDockerURL}{WFM-RC-Branch}/data_files/raincatcher-realm.json[{PROJECT_NAME} realm settings].
+
+See the {Keycloak} link:{WFM-RC-KeycloakURL}server_admin/topics/realms/create.html[realm] documentation for more information on how to
+create a new realm within a {Keycloak} instance.
+
+The realm must adhere to the following configurations in order to use it with the {PROJECT_NAME} applications
+
+* Realm settings
++
+----
+Required SSL: external
+----
+
+* Client settings
+** The realm must have three different clients for each of the {PROJECT_NAME} applications: Server, Portal and Mobile.
+** Server Client: 
++
+----
+Enabled: true
+Client Protocol: openid-connect
+Access Type: bearer-only
+Roles: admin, user
+----
+
+** Portal Client: 
++
+----
+Enabled: true
+Client Protocol: openid-connect
+Access Type: public
+Standard Flow Enabled: true
+Valid Redirect URIs: * [Ensure that this is configured with the correct URL patterns in production environments]
+Web Origins: *
+roles: admin
+----
+
+** Mobile Client:
++
+----
+Enabled: true
+Client Protocol: openid-connect
+Access Type: public
+Standard Flow Enabled: true
+Valid Redirect URIs: * [Ensure that this is configured with the correct URL patterns in production environments]
+Web Origins: *
+roles: user
```

Comment:
Same as above. We can add comment here to see "Roles section" and then document this as separate paragraph.